luaui/Widgets/gui_spectator_hud.lua

```diff
@@ -0,0 +1,2288 @@
+			return value
+		end,
+		update = nil,
+		remove = function(unitID, value)
+			return value
+		end,
+	}
+	unitCache.armyUnits = {
+		add = function(unitID, value)
+			return value[1]
+		end,
+		update = nil,
+		remove = function(unitID, value)
+			return value[1]
+		end,
+	}
+	unitCache.defenseUnits = unitCache.armyUnits
+	unitCache.utilityUnits = unitCache.armyUnits
+	unitCache.economyBuildings = unitCache.armyUnits
+
+	for _, allyID in ipairs(Spring.GetAllyTeamList()) do
+		if allyID ~= gaiaAllyID then
+			local teamList = Spring.GetTeamList(allyID)
+			for _, teamID in ipairs(teamList) do
+				unitCache[teamID] = {}
+				cachedTotals[teamID] = {}
+				unitCache[teamID].reclaimerUnits = {}
+				cachedTotals[teamID].reclaimerUnits = 0
+				unitCache[teamID].energyConverters = {}
+				cachedTotals[teamID].energyConverters = 0
+				unitCache[teamID].buildPower = {}
+				cachedTotals[teamID].buildPower = 0
+				unitCache[teamID].armyUnits = {}
+				cachedTotals[teamID].armyUnits = 0
+				unitCache[teamID].defenseUnits = {}
+				cachedTotals[teamID].defenseUnits = 0
+				unitCache[teamID].utilityUnits = {}
+				cachedTotals[teamID].utilityUnits = 0
+				unitCache[teamID].economyBuildings = {}
+				cachedTotals[teamID].economyBuildings = 0
+				local unitIDs = Spring.GetTeamUnits(teamID)
+				for i=1,#unitIDs do
+					local unitID = unitIDs[i]
+					if not Spring.GetUnitIsBeingBuilt(unitID) then
+						local unitDefID = Spring.GetUnitDefID(unitID)
+						addToUnitCache(teamID, unitID, unitDefID)
+					end
+				end
+			end
+		end
+	end
+end
+
+local function buildPlayerData()
+	playerData = {}
+	for _, allyID in ipairs(Spring.GetAllyTeamList()) do
+		if allyID ~= gaiaAllyID then
+			local teamList = Spring.GetTeamList(allyID)
+			for _,teamID in ipairs(teamList) do
+				local playerName = nil
+				local playerID = Spring.GetPlayerList(teamID, false)
+				if playerID and playerID[1] then
+					-- it's a player
+					playerName = select(1, Spring.GetPlayerInfo(playerID[1], false))
+				else
+					local aiName = Spring.GetGameRulesParam("ainame_" .. teamID)
+					if aiName then
+						-- it's AI
+						playerName = aiName
+					else
+						-- player is gone
+						playerName = "(gone)"
+					end
+				end
+
+				playerData[teamID] = {}
+				playerData[teamID].name = playerName
+
+				local teamColor = { Spring.GetTeamColor(teamID) }
+				playerData[teamID].color = teamColor
+			end
+		end
+	end
+end
+
+local function makeDarkerColor(color, factor, alpha)
+	local newColor = {}
+
+	if factor then
+		newColor[1] = color[1] * factor
+		newColor[2] = color[2] * factor
+		newColor[3] = color[3] * factor
+	else
+		newColor[1] = color[1]
+		newColor[2] = color[2]
+		newColor[3] = color[3]
+	end
+
+	if alpha then
+		newColor[4] = alpha
+	else
+		newColor[4] = color[4]
+	end
+
+	return newColor
+end
+
+local function round(num, idp)
+	local mult = 10 ^ (idp or 0)
+	return mathfloor(num * mult + 0.5) / mult
+end
+
+local function formatResources(amount, short)
```

Comment:
This can be replaced with the new `formatSI` functionality from @salinecitrine

Reply:
I'm in the camp that for different UI elements, different formatting makes sense. We have one style in top bar, other style in adv playerlist and a third style in stats panel. In my humble opinion, it's not an oversight, but rather a feature.